front_end_ang/src/app/app.component.ts

```diff
@@ -31,11 +32,7 @@ export class AppComponent implements OnInit {
 
   ngOnInit() {
     if (this.isBrowser) {
-      this.router.events.subscribe((event: any) => {
-        if (event instanceof NavigationEnd) {
-          this.updateActiveLinks();
-        }
-      });
+      this.router.events.pipe(filter(e => e instanceof NavigationEnd)).subscribe(this.updateActiveLinks);
 
```

Comment:
favor explicar

Reply:
não entendi o pq remover a autalização dos active tabs tbm

Reply:
eu passei nesse codigo pra fazer um negocio q tinha removido dps, ai eu dei uma melhorada. A funcionalidade n mudou, o pipe serve pra entrar no fluxo de dados dos eventos, o filter filtra os elementos do fluxo pra que a proxima função ignore tudo q n for do tipo NavigationEnd e o subscribe por fim executa a função this.updateActiveLinks toda vez q um evento chegar no fim desse fluxo.

Reply:
na doc do angular eles tem um exemplo assim https://angular.dev/best-practices/a11y#focus-management-after-navigation

Reply:
muito massa!!